codegen/masm/src/tests.rs

```diff
@@ -413,3 +434,470 @@ fn stackify_sum_matrix() {
     assert_eq!(stack.len(), 1);
     assert_eq!(stack.pop().map(|e| e.as_int()), Some(6));
 }
+
+#[test]
+fn i32_icmp() {
+    let mut harness = TestByEmulationHarness::default();
+
+    harness
+        .emulator
+        .load_module(
+            Box::new(
+                Module::load_intrinsic("intrinsics::i32", &harness.context.session.codemap)
+                    .expect("parsing failed"),
+            )
+            .freeze(),
+        )
+        .expect("failed to load intrinsics::i32");
+
+    let zero = Felt::new(0i32 as u32 as u64);
+    let neg_one = Felt::new(-1i32 as u32 as u64);
+    let one = Felt::new(1i32 as u32 as u64);
+    let max = Felt::new(i32::MAX as u32 as u64);
+    let min = Felt::new(i32::MIN as u32 as u64);
```

Comment:
As a side note, `proptest` should be a good fit for these kinds of tests. Its strategy for integer types covers such edge cases.

Reply:
For sure, I mainly wanted to get a few sanity tests implemented so I can unblock you on these intrinsics, but I plan to revisit many of these tests to do property-based testing :)